### PR TITLE
Bug 1074117 - Use expected conditions and optimise element lookups in UI tests app objects

### PR DESCRIPTION
**Pull request by davehunt**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[davehunt Dave Hunt (davehunt)](https://github.com/davehunt) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=de9f0fade658)

File: tests/python/gaia-ui-tests/gaiatest/apps/ui_tests_privileged/regions/user_media.py [hunk 1]
```diff
@@ -14,14 +14,20 @@ class UserMediaPage(Base):
     _frame_locator = (By.CSS_SELECTOR, "#test-iframe[src*='getusermedia']")
 
     def switch_to_frame(self):
-        self.wait_for_element_displayed(*self._frame_locator)
-        get_user_media_frame = self.marionette.find_element(*self._frame_locator)
-        self.marionette.switch_to_frame(get_user_media_frame)
+        self.wait_for_element_displ
```
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/ui_tests_privileged/regions/user_media.py]: This is obviously not correct :)
This caused failures in test_privileged_app_audio_capture_prompt.py and test_privileged_app_video_capture_prompt.py for me locally

Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/ui_tests_privileged/regions/user_media.py]: wow, how did I do that?! :P thanks for spotting it.

**Comment by davehunt** [comment]
Merged in https://github.com/mozilla-b2g/gaia/commit/0409755bd0e93d229bd43f8d5378e6dbc5e2f74c

**Pull request closed without merging**